src/components/AmountSlider.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  Button,
+  ButtonGroup,
+  Slider,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+  Stack,
+} from '@chakra-ui/react'
+import { useCallback } from 'react'
+
+import { Amount } from './Amount/Amount'
+
+type AmountSliderProps = {
+  sliderValue: number
+  handlePercentageSliderChange: (percentage: number) => void
+  handlePercentageSliderChangeEnd: (percentage: number) => void
+  onPercentageClick: (percentage: number) => void
+}
+
+export const AmountSlider: React.FC<AmountSliderProps> = ({
+  sliderValue,
+  handlePercentageSliderChange,
+  handlePercentageSliderChangeEnd,
+  onPercentageClick,
+}) => {
+  const handlePercentageClick = useCallback(
```

Comment:
prefer we just create 4 callbacks since this actually creates a callback each render instead of memoizing

Reply:
Can you show me an example, not sure what you mean here.

Reply:
this is also dumb, but prevents rerenders e.g
```
const handle25PercentClick = ...
const handle50PercentClick = ...
const handle75PercentClick = ...
const handle100PercentClick = ...
```

alternatively a whole component for each percent button but meh